types/fh-sync.d.ts

```diff
@@ -0,0 +1,314 @@
+      datasetClientCleanerCheckFrequency?: string;
+    }
+
+    /**
+     * Backoff Strategy
+     * Example: {strategy: 'exp', max: 60*1000},
+     */
+    interface PendingWorkerBackoff {
+      strategy: string;
+      max: string;
+    }
+    /**
+     * Backoff Strategy
+     * Example: {strategy: 'exp', max: 60*1000},
+     */
+    interface AckWorkerBackoff {
+      strategy: string;
+      max: string;
+    }
+
+    /**
+     * Backoff Strategy
+     * Example: {strategy: 'exp', max: 60*1000},
+     */
+    interface SyncWorkerBackoff {
+      strategy: string;
+      max: number;
+    }
+
+    type StandardCb<T> = (err: Error | string | undefined, res: T | undefined) => void;
+    type NoRespCb = (err: Error | string | undefined) => void;
+
+    /**
+     * Options used to initialize sync for specific dataset
+     */
+    interface SyncInitOptions {
+      /**
+       * Value indicating how often the dataset client should be sync with the backend. Matches the clients default 
+       * frequency. Value in seconds
+       */
+      syncFrequency?: number,
+
+      /**
+       * Value that will be used to decide if the dataset client is not active anymore.
+       */
+      clientSyncTimeout?: number,
+
+      /**
+       * Value that determines how long it should wait for the backend list operation to complete
+       */
+      backendListTimeout?: number,
+
+      /**
+      * Specify the max wait time the dataset can be scheduled to sync again after its previous schedule, in seconds.
+      */
+      maxScheduleWaitTime?: number
+    }
+
+    /**
+     * Parameters object for request and response interceptors
+     */
+    interface SyncInterceptParams {
+      query_params: any;
+      metaData: any;
+    }
+
+    /**
+     * Connect sync server to mongo and redis 
+     * 
+     * @param mongoDBConnectionUrl 
+     * @param mongoDBConnectionOption 
+     * @param redisUrl 
+     * @param cb 
+     */
+    function connect(mongoDBConnectionUrl: string, mongoDBConnectionOption: any, redisUrl: string, cb: StandardCb<void>)
+
+    /**
+     * Initialize sync for specific dataset
+     * 
+     * @param datasetId 
+     * @param options 
+     * @param callback 
+     */
+    function init(datasetId: string, options: SyncInitOptions, callback: StandardCb<void>): void;
+
+    /**
+     * Internal method used to invoke sync methods. Used to handle json request from client. 
+     * Supported operations 'sync', 'syncRecords', 'listCollisions', 'removeCollision'
+     * 
+     * @param datasetId 
+     * @param options 
+     * @param callback 
+     */
+    function invoke(datasetId: string, options: any, callback: () => void): void;
+
+    /**
+     * Stop sync loop for dataset 
+     * 
+     * @param datasetId 
+     * @param onStop callback called when operation is finished
+     */
+    function stop(datasetId: string, onStop: () => void): void;
```

Comment:
I think the onStop callback needs to be of type `NoRespCb`. (It looks like it was changed to a standard node-style callback in fh-mbaas-api 7.0?)